src/definitions/elements/step.less

```diff
@@ -196,53 +324,107 @@
+    .ui.vertical.steps:not(.circular) .step {
         justify-content: flex-start;
         border-radius: @borderRadius;
         padding: @verticalPadding @horizontalPadding;
         border-right: none;
         border-bottom: @verticalDivider;
     }
-    .ui.vertical.steps .step:first-child {
+    .ui.vertical.steps:not(.circular) .step:first-child {
         padding: @verticalPadding @horizontalPadding;
         border-radius: @stepsBorderRadius @stepsBorderRadius 0 0;
     }
-    .ui.vertical.steps .step:last-child {
+    .ui.vertical.steps:not(.circular) .step:last-child {
         border-bottom: none;
         border-radius: 0 0 @stepsBorderRadius @stepsBorderRadius;
     }
-    .ui.vertical.steps .step:only-child {
+    .ui.vertical.steps:not(.circular) .step:only-child {
         border-radius: @stepsBorderRadius;
     }
 
     /* Arrow */
-    .ui.vertical.steps .step::after {
+    .ui.vertical.steps:not(.circular) .step::after {
         top: @verticalArrowTopOffset;
         right: @verticalArrowRightOffset;
         border-width: @verticalArrowBorderWidth;
         display: @verticalArrowDisplay;
     }
-    .ui.right.vertical.steps .step::after {
+    .ui.right.vertical.steps:not(.circular) .step::after {
         border-width: @verticalLeftArrowBorderWidth;
         left: @verticalLeftArrowLeftOffset;
         right: @verticalLeftArrowRightOffset;
         transform: translateY(-50%) translateX(-50%) rotate(-45deg);
     }
 
-    .ui.vertical.steps .active.step::after {
+    .ui.vertical.steps:not(.circular) .active.step::after {
         display: @verticalActiveArrowDisplay;
     }
-    .ui.vertical.steps .step:last-child::after {
+    .ui.vertical.steps:not(.circular) .step:last-child::after {
         display: @verticalLastArrowDisplay;
     }
-    .ui.vertical.steps .active.step:last-child::after {
+    .ui.vertical.steps:not(.circular) .active.step:last-child::after {
         display: @verticalActiveLastArrowDisplay;
     }
+
+    /* Circular Step */
+    & when (@variationCircularSteps) {
+        .ui.circular.vertical.steps {
+            display: inline-flex;
+            flex-flow: column wrap;
+            align-items: flex-start;
+
+            & .step {
+                position: relative;
+                width: 100%;
+
+                &::before {
+                    left: 0;
+                    top: 0;
+                }
+
+                &::after {
+                    left: @verticalCircularStepIconDistance;
+                    top: @verticalCircularStepIconDistance;
+                }
+
+                & .content {
+                    padding-left: @verticalCircularStepContentPadding;
+                }
+
+                &:not(:last-child) {
+                    padding-bottom: @verticalCircularStepPadding;
+
+                    & .content::before {
+                        content: "";
+                        position: absolute;
+                        border-left: @circularStepBorder;
```

Comment:
An idea:
What if we make connection line optional as `ui connected circular steps`? This is how we already have it for `ui connected feed`
So only the border-left and each border-color setting would get an additional `connected` selector

What do you think?

Reply:
Do you mean only for vertical step? Then, that's a good idea. 

But for the horizontal step, if the connection line isn't showing, then the UI looks awkward.
![horizontal-step-with-no-connection](https://github.com/fomantic/Fomantic-UI/assets/930315/d2686d28-37b5-4a13-89aa-00cedd7ecc0d)

In this case, we would need title and description like vertical step. And this would be a problem when the screen size is narrower.